### ISSUE TITLE
HNC: remove the long delays in object_controller_test.go

### ISSUE THREAD

**Issue by adrianludwin** [kubernetes-sigs/multi-tenancy#101]
The object propagation tests have 5s delays, and even then they sometimes fail. This doesn't really make sense; they should be passing reliably. We should figure out what's causing the failures in these (very small) test cases and fix them.

**Comment by adrianludwin** [comment]
/assign @lafh 

The code and tests are both by @yiqigao217 so maybe sync with her? But she's got a lot going on right now and could probably use the debugging help on this one.

**Comment by adrianludwin** [comment]
/assign @yiqigao217 

I believe you're having a closer look at this with your latest PR?

**Comment by yiqigao217** [comment]
Should we close this issue since now we are using conditions instead of annotations? And the time-out issue no longer happens when detecting the object conditions within the default 2s.

Previously, the issue was that it took up to (or more than) 5s to detect that the modified annotation is set on object.

**Comment by adrianludwin** [comment]
Yeah makes sense. If you have any idea _why_ this problem went away it
would be nice to write that down, but if we don't, then I guess that's ok
too.

**Comment by yiqigao217** [comment]
UPDATE: When I was trying to figure out why, I started by testing the old annotation version (current upstream/master), it passed. Even after I removed the 5s and thus the time-out limit is set back to the default 2s, it passed without being flaky (I ran `make test` for more than 10 times).

I guess the previous issue was on the forest lock & calling API server, since the current upstream version is the one after we have consolidated all forest lock calls.
